src/theory/theory_model_builder.cpp

```diff
@@ -436,6 +526,148 @@ bool TheoryEngineModelBuilder::buildModel(Model* m)
     }
   }
 
+  Trace("model-builder") << "Compute assignable information..." << std::endl;
+  // The set of equivalence classes that are "assignable", i.e. those that
+  // have an assignable expression in them (see isAssignable), and
+  // have not already been assigned.
+  std::unordered_set<Node, NodeHashFunction> assignableEqc;
+  // The set of equivalence classes that are "evaluable", i.e. those that
+  // have an expression in them that is not assignable, and have not already
+  // been assigned.
+  std::unordered_set<Node, NodeHashFunction> evaluableEqc;
+  // Assigner objects for relevant equivalence classes
+  std::map<Node, Assigner> eqcToAssigner;
+  // Maps equivalence classes to the equivalence class that maps to its assigner
+  // object in the above map.
+  std::map<Node, Node> eqcToAssignerMaster;
+  // compute the above information
+  {
+    bool computeAssigners = tm->hasAssignmentExclusionSets();
+    std::unordered_set<Node, NodeHashFunction> processed;
+    eqcs_i = eq::EqClassesIterator(ee);
+    bool assignable = false;
+    bool evaluable = false;
```

Comment:
Document these variables.  There seem to be several invariants connected with each of them.